File: src/main/java/com/assessment/comsc/panelStaff/PanelStaff.java
package com.assessment.comsc.panelStaff;

import lombok.AllArgsConstructor;
import lombok.Data;

@Data
@AllArgsConstructor
public class PanelStaff {
    private Boolean considered;
    private String considerationDate;
    private String feedback;
    private Boolean notifiedModeratorSetter;
    private String notificationDate;
    private Boolean notifiedModuleLeader;
    private String moduleLeaderNotificationDate;
    private String assessmentStatus;
    private Integer assessmentId;

    public Boolean getConsidered() {
        return considered;
    }

    public String getConsiderationDate() {
        return considerationDate;
    }

    public String getFeedback() {
        return feedback;
    }

    public Boolean getNotifiedModeratorSetter() {
        return notifiedModeratorSetter;
    }

    public String getNotificationDate() {
        return notificationDate;
    }

    public Boolean getNotifiedModuleLeader() {
        return notifiedModuleLeader;
    }

    public String getModuleLeaderNotificationDate() {
        return moduleLeaderNotificationDate;
    }

    public String getAssessmentStatus() {
        return assessmentStatus;
    }

    public Integer getAssessmentId(){
        return assessmentId;
    }
}

File: src/main/resources/static/js/panelStaff.js
$(document).ready(function() {
    $("#saveFeedback").click(function(e){
        var considered = $('#considered').prop('checked');
        var considerationDate = $('#considerationDate').val();
        var feedback = $('#feedback').val();
        var notifiedModeratorSetter = $('#notifiedModeratorSetter').prop('checked');
        var notificationDate = $('#notificationDate').val();
        var notifiedModuleLeader = $('#notifiedModuleLeader').prop('checked');
        var moduleLeaderNotificationDate = $('#moduleLeaderNotificationDate').val();
        var assessmentStatus = $("#assessmentStatus").val();
        var jsonData = {
            considered: considered,
            considerationDate: considerationDate,
            feedback: feedback,
            notifiedModeratorSetter: notifiedModeratorSetter,
            notificationDate: notificationDate,
            notifiedModuleLeader: notifiedModuleLeader,
            moduleLeaderNotificationDate: moduleLeaderNotificationDate,
            assessmentStatus: assessmentStatus,
            assessmentId: decodeURIComponent(new URLSearchParams(window.location.search).get("id")) || 0
        };
        $.ajax({
            type: 'POST',
            url: '/public/saveFeedback',
            contentType: 'application/json',
            data: JSON.stringify(jsonData),
            success: function () {
                alert("Save successful");
                window.history.back();
            },
            error: function (xhr, status, error) {
                console.error("Save failed. Status: " + status + ", Error: " + error);
                alert("Save failed");
            }
        });

    });
})
